quill-core/src/main/scala/io/getquill/norm/capture/Dealias.scala

```diff
@@ -36,8 +36,8 @@ case class Dealias(state: Option[Ident]) extends StatefulTransformer[Option[Iden
         val (bn, _) = apply(b)
         (UnionAll(an, bn), Dealias(None))
       case Join(t, a, b, iA, iB, o) =>
-        val ((an, iAn, on), ont) = dealias(a, iA, o)((_, _, _))
-        val ((bn, iBn, onn), _) = ont.dealias(b, iB, on)((_, _, _))
+        val ((an, iAn, on), _) = dealias(a, iA, o)((_, _, _))
+        val ((bn, iBn, onn), _) = dealias(b, iB, on)((_, _, _))
```

Comment:
Could you give more context on these changes? I was debugging #939 and found that the root issue comes from this class. (this comment does not relate to spark module)

Reply:
This was another bug actually. It was using the same alias for the case where the same entity is on both sides of the join `(qr1).join(qr1)`. Each branch of the join should be treated as independent, so it doesn't make sense to use the transformer of the first one.